lib/utils/logDeprecation.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+const chalk = require('chalk');
+const memoizee = require('memoizee');
+
+const disabledCodesByEnv = extractCodes(process.env.SLS_DEPRECATION_DISABLE);
+
+const loggedDeprecations = new Set();
+
+function extractCodes(codesStr) {
+  if (!codesStr) {
+    return new Set();
+  }
+  return new Set(codesStr.split(','));
+}
+
+const resolveDeprecatedByService = memoizee((serviceConfig = {}) => {
+  let disabledDeprecations = [];
+  if (typeof serviceConfig.disabledDeprecations === 'string') {
+    disabledDeprecations = [serviceConfig.disabledDeprecations];
+  } else {
+    disabledDeprecations = serviceConfig.disabledDeprecations || [];
+  }
+  return new Set(disabledDeprecations);
+});
+
+function writeDeprecation(code, message) {
+  process.stdout.write(
+    [
+      `Serverless: ${chalk.red(`Deprecation Warning: ${message}`)}`,
+      `            ${chalk.dim(
+        `More Info: https://www.serverless.com/framework/docs/deprecations/#${code}`
+      )}}`,
+    ].join('\n')
+  );
```

Comment:
Let's also add new line at the end (in case of `process.stdout.write` they're not added as it's with `console.log`

Reply:
Done